contracts/bonding/BondingManager.sol

```diff
@@ -1134,29 +1195,49 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
     ) internal view returns (uint256 cStake, uint256 cFees) {
         // Fetch start cumulative factors
         EarningsPool.Data memory startPool = cumulativeFactorsPool(_transcoder, _startRound);
+        // Fetch end cumulative factors
+        EarningsPool.Data memory endPool = latestCumulativeFactorsPool(_transcoder, _endRound);
 
+        return delegatorCumulativeStakeAndFees(startPool, endPool, _stake, _fees);
+    }
+
+    /**
+     * @notice Calculates a delegator's cumulative stake and fees using the LIP-36 earnings claiming algorithm.
+     * @dev This is a mostly a memroy-only function to be called from other contracts. Created for historical stake
+     * calculations with BondingCheckpoints.
+     * @param _startPool The earning pool from the start round for the start cumulative factors. Normally this is the
+     * earning pool from the {Delegator-lastclaimRound}+1 round, as the round where `bondedAmount` was measured.
+     * @param _endPool The earning pool from the end round for the end cumulative factors
+     * @param _stake The delegator initial stake before including earned rewards. Normally the {Delegator-bondedAmount}
+     * @param _fees The delegator's initial fees before including earned fees
+     * @return cStake , cFees where cStake is the delegator's cumulative stake including earned rewards and cFees is the
+     * delegator's cumulative fees including earned fees
+     */
+    function delegatorCumulativeStakeAndFees(
```

Comment:
Should we move this memory/lib version of the func to `EarningsPool(LIP36)?` perhaps?

Reply:
Done as it made a lot of sense for testability as well. I didn't want to mock this function just because it happened to be in the `BondingManager`. It makes a lot of conceptual sense as well, since it implements the logic defined in LIP-36, so all the more sense for it to be in `EarningsPoolLIP36`.